### PR TITLE
Handle SET 'param' TO 'value' statement

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

File: sql/src/main/java/io/crate/action/sql/SQLOperations.java [hunk 1]
```diff
@@ -136,7 +133,7 @@ public Session createSession(@Nullable String defaultSchema) {
         private Throwable throwable = null;
         private List<List<Object>> bulkParams = new ArrayList<>();
         private List<ResultReceiver> resultReceivers = new ArrayList<>();
-        private Map<String, String> settings;
+        private Map<String, String> settings = new HashMap<>();
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/SQLOperations.java]: Can you initialize this to an Collections.emptyMap and in the applySettings just change the assignment to the settings of the Set statements (maybe with a unmodifiableMap protection) ?

Comment (matriv) [inline sql/src/main/java/io/crate/action/sql/SQLOperations.java]: The idea behind putAll is that a SET [SESSION] param=value can happen at any time within the session so we should update the params.

Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/SQLOperations.java]: Ah okay, good point.

**Comment by mfussenegger** [comment]
lgtm

**Pull request merged**